File: C++/source/Framework/Messaging/LightHouse/LightPulseRegistrationMessage.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightPulseRegistrationMessage.h"
#include "Framework/Messaging/LightHouse/LightPulseDispatchObjectManager.h"
#include "LightPulseDispatchTypes.h"

namespace WaveNs
{

LightPulseRegistrationMessage::LightPulseRegistrationMessage ()
    : WaveMessage (LightPulseDispatchObjectManager::getWaveServiceId (), LIGHT_PULSE_DISPATCH_REGISTER_FOR_LIGHT_PULSE)
{
}

LightPulseRegistrationMessage::~LightPulseRegistrationMessage ()
{
}

string LightPulseRegistrationMessage::getLightPulseName () const
{
    return (m_lightPulseName);
}

void LightPulseRegistrationMessage::setLightPulseName (const string &lightPulseName)
{
    m_lightPulseName = lightPulseName;
}

void LightPulseRegistrationMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization ();

    addSerializableAttribute (new AttributeString (&m_lightPulseName, "lightPulseName"));
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightPulseRegistrationMessage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

import com.CxWave.Wave.Framework.Messaging.Local.WaveMessage;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.Message;

@Message (om = LightPulseDispatchObjectManager.class)
public class LightPulseRegistrationMessage extends WaveMessage
{
    private String m_lightPulseName;

    public LightPulseRegistrationMessage ()
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_REGISTER_FOR_LIGHT_PULSE);
    }

    public LightPulseRegistrationMessage (final String lightPulseName)
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_REGISTER_FOR_LIGHT_PULSE);

        m_lightPulseName = lightPulseName;
    }

    public String getLightPulseName ()
    {
        return m_lightPulseName;
    }

    public void setLightPulseName (final String lightPulseName)
    {
        m_lightPulseName = lightPulseName;
    }
}
